torch/csrc/api/include/torch/nn/module.h

```diff
@@ -0,0 +1,130 @@
+#pragma once
+
+#include <torch/tensor.h>
+
+#include <memory>
+#include <string>
+#include <unordered_map>
+#include <vector>
+
+namespace at {
+enum class ScalarType;
+} // namespace at
+
+namespace torch { namespace nn {
+
+struct Archive;
+struct BufferCursor;
+struct ModuleCursor;
+struct ParameterCursor;
+
+/// The base class for all torch modules.
+class Module {
+ public:
+  /// Tells the base `Module` about the name of the submodule.
+  explicit Module(std::string name);
+  virtual ~Module();
+
+  /// Returns the name of the `Module`.
+  const std::string& name() const noexcept;
+
+  /// Performs a recursive clone of the entire module hierarchy. This is to
```

Comment:
does it clone parameters too?

Reply:
Yeah it should be like `copy.deepcopy(module)` in Python

Reply:
Can you please mention that in the comment? It's quite important